src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -110,6 +139,21 @@
         (.put puppet-config dir (fs/absolute-path value))))
     puppet-config))
 
+(schema/defn borrow-with-timeout-fn :- jruby-schemas/JRubyPuppetBorrowResult
```

Comment:
Don't worry about spending any time tracking this down and potentially changing it unless it's worthwhile.  I'm more just curious